templates/components/itilobject/timeline/form_document_item.html.twig

```diff
@@ -9,11 +9,30 @@
          {% set filename = entry_i['filename'] ?? entry_i['name'] %}
 
          {% if entry_i['filename'] %}
+            {% set docpath = path("front/document.send.php?docid=" ~ entry_i['id']) %}
+            {% set is_image = isImage(entry_i['filepath']) %}
             <div class="col text-truncate">
-               <a href="{{ path("front/document.send.php?docid=" ~ entry_i['id']) }}" target='_blank'>
-                  <img src="{{ getIconForFilename(filename) }}" alt="{{ __("File extension") }}">
-                  {{ name }}
-               </a>
+               {% if is_image %}
+                  {% set img_size = getImageSize(entry_i['filepath']) %}
+                  {% include 'components/photoswipe.html.twig' with {
+                     'imgs': [{
+                        'title': '',
+                        'thumbnail_src': docpath ~ '&context=timeline',
+                        'thumbnail_w': 'auto',
+                        'thumbnail_h': 'auto',
+                        'src': docpath,
+                        'w': img_size[0],
+                        'h': img_size[1],
+                        'clearable': false,
+                     }]
+                  } %}
+                  <a href="{{ docpath }}" target='_blank'>{{ name }}</a>
```

Comment:
one thing here, in case of picture, i'm not sure having a plain text under is very usefull.
If needed, i think a title attribute should be sufficient (you can add tootltip toggle also if needed)